### PR TITLE
Display GPU requests and limits

### PR DESCRIPTION
**Pull request by mYmNeo**
Signed-off-by: thomassong <thomassong@tencent.com>

**What this PR does / why we need it**:
Display GPU requests and limits in `kubectl describe`, like this
```
Non-terminated Pods:		(1 in total)
  Namespace			Name					CPU Requests	CPU Limits	Memory Requests	Memory Limits	GPU Requests	GPU Limits
  ---------			----					------------	----------	---------------	-------------	------------	----------
  kube-system			kube-dns-806549836-45kg7		260m (6%)	0 (0%)		110Mi (1%)	170Mi (2%)	0 (0%)		0 (0%)
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
 CPU Requests	CPU Limits	Memory Requests	Memory Limits	GPU Requests	GPU Limits
  ------------	----------	---------------	-------------	------------	----------
  260m (6%)	0 (0%)		110Mi (1%)	170Mi (2%)	0 (0%)		0 (0%)
```

### CODE REVIEW

File: pkg/printers/internalversion/describe.go [hunk 1]
```diff
@@ -2407,8 +2407,8 @@ func (d *HorizontalPodAutoscalerDescriber) Describe(namespace, name string, desc
 
 func describeNodeResource(nodeNonTerminatedPodsList *api.PodList, node *api.Node, w PrefixWriter) error {
 	w.Write(LEVEL_0, "Non-terminated Pods:\t(%d in total)\n", len(nodeNonTerminatedPodsList.Items))
-	w.Write(LEVEL_1, "Namespace\tName\t\tCPU Requests\tCPU Limits\tMemory Requests\tMemory Limits\n")
-	w.Write(LEVEL_1, "---------\t----\t\t------------\t----------\t---------------\t-------------\n")
+	w.Write(LEVEL_1, "Namespace\tName\t\tCPU Requests\tCPU Limits\tMemory Requests\tMemory Limits\tGPU Requests\tGPU Limits\n")
```
Comment (cmluciano) [inline pkg/printers/internalversion/describe.go]: I don't think that we support GPU requests, only limits

File: pkg/printers/internalversion/describe.go [hunk 2]
```diff
@@ -2420,29 +2420,38 @@ func describeNodeResource(nodeNonTerminatedPodsList *api.PodList, node *api.Node
...
 		fractionMemoryReq := float64(memoryReq.Value()) / float64(allocatable.Memory().Value()) * 100
 		fractionMemoryLimit := float64(memoryLimit.Value()) / float64(allocatable.Memory().Value()) * 100
-		w.Write(LEVEL_1, "%s\t%s\t\t%s (%d%%)\t%s (%d%%)\t%s (%d%%)\t%s (%d%%)\n", pod.Namespace, pod.Name,
+		fractionGPUReq := float64(gpuReq.Value()) / float64(allocatable.NvidiaGPU().Value()) * 100
+		fractionGPULimit := float64(gpuLimit.Value()) / float64(allocatable.NvidiaGPU().Value()) * 100
+		w.Write(LEVEL_1, "%s\t%s\t\t%s (%d%%)\t%s (%d%%)\t%s (%d%%)\t%s (%d%%)\t%s (%d%%)\t%s (%d%%)\n", pod.Namespace, pod.Name,
 			cpuReq.String(), int64(fractionCpuReq), cpuLimit.String(), int64(fractionCpuLimit),
-			memoryReq.String(), int64(fractionMemoryReq), memoryLimit.String(), int64(fractionMemoryLimit))
+			memoryReq.String(), int64(fractionMemoryReq), memoryLimit.String(), int64(fractionMemoryLimit),
+			gpuReq.String(), int64(fractionGPUReq), gpuLimit.String(), int64(fractionGPULimit))
```
Comment (cmluciano) [inline pkg/printers/internalversion/describe.go]: Fractional GPU requests are not support, nor will the probably be supported for a long time.

Comment (cmluciano) [inline pkg/printers/internalversion/describe.go]: Nit: Int64 seems unnecessary since only whole numbers are supported at the moment.

**Review by cmluciano — CHANGES_REQUESTED** [review]
I do not believe that GPU requests will provide useful information since limits are not supported. 

Some of the type declarations do not feel appropriate for the current GPU implementation.

**Comment by dims** [comment]
@k8s-bot ok to test

File: pkg/printers/internalversion/describe.go (hunk 1 above)
Comment (mYmNeo) [inline pkg/printers/internalversion/describe.go]: The document says it should be set by using limits, but actually k8s help to [complete the requests](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/defaults.go#L126) and give a guarantee of [check](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/validation/validation.go#L50)

File: pkg/printers/internalversion/describe.go (hunk 2 above)
Comment (mYmNeo) [inline pkg/printers/internalversion/describe.go]: Yeah, I know. But there'll be no need to change the code again if we support share the GPU in the future?

**Pull request closed without merging**